cunumeric/deferred.py

```diff
@@ -1132,6 +1132,40 @@ def dot(self, src1, src2, stacklevel=0, callsite=None):
                 f"dot between {rhs1_array.ndim}d and {rhs2_array.ndim}d arrays"
             )
 
+    # Create array from input array and indices
+    @profile
+    @auto_convert([1])
+    @shadow_debug("choose", [1])
```

Comment:
Similarly to the comment above, `shadow_debug` won't work for this method, but this is not needed for correctness in normal execution (and I think this feature is going out the door soon), so I would just remove this line.